File: src/test/java/com/dsm/pick/domains/service/JwtServiceTest.java
//package com.dsm.pick.domains.service;
//
//import io.jsonwebtoken.Jwt;
//import io.jsonwebtoken.SignatureAlgorithm;
//import org.junit.jupiter.api.Test;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.beans.factory.annotation.Value;
//
//import javax.crypto.spec.SecretKeySpec;
//import javax.xml.bind.DatatypeConverter;
//import java.security.Key;
//import java.util.Date;
//
//import static org.junit.jupiter.api.Assertions.*;
//
//class JwtServiceTest {
//
//    JwtService jwtService = new JwtService("dhwlddjgmanf");
//
//    @Test
//    void createAccessToken() {
//        String teacherId = "kiwi";
//        String accessToken = jwtService.createAccessToken(teacherId);
//    }
//
//    @Test
//    void createRefreshToken() {
//        String teacherId = "lemon";
//        String refreshToken = jwtService.createRefreshToken(teacherId);
//    }
//
//    @Test
//    void getTeacherId() {
//        String teacherId = "richie";
//        String accessToken = jwtService.createAccessToken(teacherId);
//
//        String findTeacherId = jwtService.getTeacherId(accessToken);
//
//        assertEquals(teacherId, findTeacherId);
//    }
//
//    @Test
//    void getExpiration() {
//        String teacherId = "quince";
//        String accessToken = jwtService.createAccessToken(teacherId);
//        Date expiration = jwtService.getExpiration(accessToken);
//    }
//
//    @Test
//    void isUsableToken() {
//        String teacherId = "jujube";
//        String accessToken = jwtService.createAccessToken(teacherId);
//
//        boolean isValid = jwtService.isValid(accessToken);
//        boolean isTimeOut = jwtService.isNotTimeOut(accessToken);
//
//        assertTrue(isValid && isTimeOut);
//    }
//}

File: src/test/kotlin/com/dsm/pick/service/JwtServiceTest.kt
package com.dsm.pick.service

import org.assertj.core.api.Assertions.*
import org.junit.jupiter.api.Test

internal class JwtServiceTest () {
    private val jwtService = JwtService("kotlin-love")

    @Test
    fun `토큰 생성 OK`() {
        val accessToken = jwtService.createToken("teacherId", Token.ACCESS)
        val refreshToken = jwtService.createToken("teacherId", Token.REFRESH)

        assertThat(accessToken).isNotBlank
        assertThat(refreshToken).isNotBlank
    }

    @Test
    fun `유효한 토큰인지 검사 OK`() {
        val accessToken = jwtService.createToken("teacherId", Token.ACCESS)
        val isValidationToken = jwtService.isValid(accessToken)

        assertThat(isValidationToken).isTrue
    }

    @Test
    fun `토큰에서 teacherId 추출 OK`() {
        val accessToken = jwtService.createToken("teacherId", Token.ACCESS)
        val teacherId = jwtService.getTeacherId(accessToken)

        assertThat(teacherId).isEqualTo("teacherId")
    }
}